server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -901,6 +930,16 @@ protected void abandonSegment(final long truncatedTime, final Sink sink)
           sink.getVersion(),
           new SingleElementPartitionChunk<>(sink)
       );
+      Iterables.transform(sink, new Function<FireHydrant, Void>()
+      {
+        @Nullable
+        @Override
+        public Void apply(@Nullable FireHydrant input)
+        {
+          cache.close(input.getSegment().getIdentifier() + "_" + input.getCount());
+          return null;
+        }
+      });
```

Comment:
do we need to iterate for the close to be called ? or may be simply replace it with a for loop. 
Also, can we abstract the the namespace creation method.